src/server/build-server.ts

```diff
@@ -67,36 +71,76 @@ export function buildServer(): FastifyInstance {
       });
     }
   );
+  app.post(
+    "/auth",
+    PostAuthBodySchema,
+    async (
+      request: FastifyRequest<{ Body: GetUserRequestBody }>,
+      reply: FastifyReply
+    ) => {
+      const { email, password, rememberMe } = request.body;
+      const user = await prisma.user.findUnique({
+        where: { email },
+      });
 
-  app.post("/auth", AuthRequestBodySchema, async (request, reply) => {
-    const { email, password, rememberMe } = request.body;
-    const user = await prisma.user.findUnique({
-      where: { email },
-    });
-
-    if (!user) {
-      throw new CustomError("Email not registered on platform", "EML_02");
-    } else {
-      const isPasswordValid = await compare(password, user.password);
-      if (!isPasswordValid) {
-        throw new CustomError("Wrong password. Try again", "PSW_03");
+      if (!user) {
+        throw new CustomError("Email not registered on platform", "EML_02");
+      } else {
+        const isPasswordValid = await compare(password, user.password);
+        if (!isPasswordValid) {
+          throw new CustomError("Wrong password. Try again", "PSW_03");
+        }
       }
+
+      const token = jwt.sign({ id: user.id }, process.env.TOKEN_KEY, {
+        expiresIn: rememberMe ? "1w" : Number(process.env.TOKEN_TIMEOUT) || 30,
+      });
+
+      return reply.code(200).send({
+        user: {
+          id: user.id,
+          name: user.name,
+          email: user.email,
+          birthDate: user.birthDate,
+        },
+        token: token,
+      });
     }
+  );
 
-    const token = jwt.sign({ id: user.id }, process.env.TOKEN_KEY, {
-      expiresIn: rememberMe ? "1w" : Number(process.env.TOKEN_TIMEOUT) || 30,
-    });
+  app.get(
+    "/users/:id",
+    async (request: FastifyRequest<{ Params: { id: string } }>, reply) => {
+      isAuthenticated(request);
+      const { id } = request.params;
+
+      const userId = Number(id);
+      if (isNaN(userId)) {
+        throw new CustomError(
+          "Invalid ID. User ID must be a number",
+          "USR_02",
+          "The user ID must be an integer"
+        );
+      }
+
+      const user = await prisma.user.findUnique({
+        where: { id: Number(id) },
```

Comment:
```suggestion
        where: { id: userId },
```

Vc já fez essa opração `Number(id)`